### ISSUE TITLE
Migration to gitlab

### ISSUE THREAD

**Issue by FascinatedBox** [FascinatedBox/lily#377]
You've probably heard by now how GitHub was bought out by Microsoft. Before that, GH was a company by itself, so it had no explicit ties to any particular company over another. It uses Ruby to my knowledge, but otherwise doesn't promote and one PL over another. It's neutral, and that's good.

I avoided putting this issue in immediately after GH was bought out. I don't like jumping on bandwagons. For a project that has as many eyes on it as this does, I especially don't want to seem as though I'm jumping on trends. People like stability and predictability with programming languages, I think.

I'd like to clarify, before going further, that my complaint is not that the acquisition was done by Microsoft. I'm a long-time Linux user, but I'm not the 'I hate M$' type. I'm against the acquisition because of it being by a large company that's already getting hooks into a lot of other places. It's a teensy bit political.

GitLab is the one I keep hearing mentioned. There are some positives I like about it, like being able to run different workers. I like GitLab trying to be a whole system of CI/CD since there are analytics I'd like to know about the interpreter: Questions like "is this commit causing more memory to be used", "what's the parse time of various constructs, is it getting faster/slower with this", and so on.

My plan for the migration is as follows:

I'll start by creating mirror repositories for Lily itself and all of the repositories that I currently have. The linkages will be swapped over from pointing to GitHub to pointing to GitLab, once I discover how to migrate over gh-pages.

Once the migration is done, I'll point the repositories I have on this site to those over on GitLab. The GitLab repositories will cease to be updated. I'm considering consolidating the Discord and irc channels down to GitLab's Mattermost, but I need to get a feel for Mattermost first to see if I want to do that. I won't do that before the GH -> GL migration is fully done.

I also apologize for having been away for a while. Some of that was due to trying to sort myself out, and some of it was from not really knowing what I want to do with this thing I've built.

**Comment by FascinatedBox** [comment]
All other Lily repositories now point to their gitlab equivalents and are archived. I'm going to close the other issues, then this one, and finish by archiving this repository.
